plugins/meteor_engage_warn.py

```diff
@@ -75,9 +77,22 @@ def _display_meteor_pos_warning_msg(self, end_pos: Dict[str, float]) -> bool:
         pos_str = "\n". join(pos_str)
 
         # Guess (back) which position the users wants to go to
-        target_pos = self.main_app.main_frame.getCurrentPostureLabel(end_pos)
+        target_pos = self.main_app.main_data.posture_manager.getCurrentPostureLabel(end_pos)
+        stage = self.main_app.main_data.get_component("stage")
```

Comment:
Does this work?! Did you test this plugin? I don't see any `get_component()` method.
Typically it should be:
`stage = self.main_app.main_data.stage`